packages/auth/src/sessionStore.ts

```diff
@@ -66,6 +66,9 @@ export class InstanceDataStore extends SessionDataStore {
  */
 export class LocalStorageStore extends SessionDataStore {
   key: string;
+  // sessionData points to localStorage instance for sharing session between parallel calls of put or delete file
+  // To prevent the conflicting issue of etags deletion or updation
+  sessionData: SessionData = new SessionData({});
```

Comment:
It's not clear to me that storing/caching more state in memory is the solution here, seems like it would lead to more kinds of race conditions.

It makes more sense to me that the functions modifying session data (like mutating etags) should have fixes to avoid caching and writing old `SessionData` between network calls. E.g. the deleteFile function should get the latest session data before updating and writing:
```ts
await deleteFromGaiaHub(path, gaiaHubConfig);
// get latest session data which could have been modifying while the above `await` was resolving
const sessionData = this.userSession.store.getSessionData();
delete sessionData.etags![path];
this.userSession.store.setSessionData(sessionData);
```

Reply:
The `deleteFile` function call is already getting the latest `session ` values. I don't think we can even touch the deleteFile function to solve this issue. It's a concurrency issue that only happens when you call deleteFile without await statement and in my opinion making session a shared variable solves the problem. And i also provided `script` in the description of this pr to prove the correct behaviour.

Secondly other functions that are modifying the etags are transparent similar to `deleteFile` as did not make a change in `deleteFile` function.  Hope it make sense.
cc: @zone117x

Reply:
> The deleteFile function call is already getting the latest session values

That isn't correct, here's the code:
https://github.com/blockstack/stacks.js/blob/9816b3c7ae4e9a8f7fcff219239f8b6952f70471/packages/storage/src/storage.ts#L656-L658

Reply:
@zone117x  Yeah that's what i fixed.

Reply:
I'd just re-iterate my first comment above that I don't think this is a good approach, and that a more robust fix would be ensuring the localStorage `sessionData` is retrieved, updated, and saved _after_ the async network calls.

Reply:
@zone117x I did full research in terms of approach to get this issue fixed. You may mention any other good approach if you would like. Secondly i would like you to reconsider the behaviour in this issue. The user is facing this issue because he want to deleteFiles in a parallel fashion without `await` statement. Considering that the best approach is to make a shared variable that will remain in `sync` with localstorage all time, that's what i did. It solved the issue. I also provided the testcases and `scripts` to verify the results.

Reply:
@zone117x Alright
- Would you mind sharing many ways of caching localStorage data in memory
- Can you produce what issues this solution will create
- Can you share the solution that you mentioned as pretty straightforward 

I am ready to implement if there are better solutions. Please guide me in the right direction.

Reply:
Yes, test something like the code snippet I shared in my first comment:
```ts
await deleteFromGaiaHub(path, gaiaHubConfig);
// get latest session data which could have been modified while the above `await` was resolving
const sessionData = this.userSession.store.getSessionData();
delete sessionData.etags![path];
this.userSession.store.setSessionData(sessionData);
```

Reply:
Tested and added this approach. Thanks.

Reply:
Looks good, thanks!